crux-lucene/src/crux/lucene.clj

```diff
@@ -0,0 +1,187 @@
+                     (for [eid (db/ave index-snapshot a v nil entity-resolver-fn)]
+                       [(db/decode-value index-snapshot eid) v a score]))))
+         (normalise-scores)
+         (into []))))
+
+(defmethod q/pred-args-spec 'text-search [_]
+  (s/cat :pred-fn  #{'text-search} :args (s/spec (s/cat :v string? :attr (s/? keyword?))) :return (s/? :crux.query/binding)))
+
+(defmethod q/pred-constraint 'text-search [_ {:keys [encode-value-fn idx-id arg-bindings rule-name->rules return-type tuple-idxs-in-join-order] :as pred-ctx}]
+  (let [[vval attr] (rest arg-bindings)]
+    (fn pred-get-attr-constraint [index-snapshot {:keys [entity-resolver-fn] :as db} idx-id->idx join-keys]
+      (q/bind-binding return-type tuple-idxs-in-join-order (get idx-id->idx idx-id) (full-text *node* index-snapshot entity-resolver-fn attr vval)))))
+
+(defn- entity-txes->content-hashes [txes]
+  (set (for [^EntityTx entity-tx txes]
+         (.content-hash entity-tx))))
+
+(defrecord LuceneDecoratingIndexStore [index-store document-store lucene-node]
+  db/IndexStore
+  (index-docs [this docs]
+    (db/index-docs index-store docs))
+  (unindex-eids [this eids]
+    (try
+      (evict! index-store lucene-node eids)
+      (catch Throwable t
+        (clojure.tools.logging/error t)
+        (throw t)))
+    (db/unindex-eids index-store eids))
+  (index-entity-txs [this tx entity-txs]
+    (let [{:keys [^Directory directory ^Analyzer analyzer]} lucene-node
+          index-writer (IndexWriter. directory, (IndexWriterConfig. analyzer))]
+      (try
+        (let [content-hashes (entity-txes->content-hashes entity-txs)
+              docs (vals (db/fetch-docs document-store content-hashes))]
+          (write-docs! index-writer docs)
+          (db/index-entity-txs index-store tx entity-txs)
+          (.close index-writer))
+        (catch Throwable t
+          (clojure.tools.logging/error t)
+          (.rollback index-writer)
+          (throw t)))))
+  (mark-tx-as-failed [this tx]
+    (db/mark-tx-as-failed index-store tx))
+  (store-index-meta [this k v]
+    (db/store-index-meta index-store k v))
+  (read-index-meta [this k]
+    (db/read-index-meta index-store k))
+  (read-index-meta [this k not-found]
+    (db/read-index-meta index-store k not-found))
+  (latest-completed-tx [this]
+    (db/latest-completed-tx index-store))
+  (tx-failed? [this tx-id]
+    (db/tx-failed? index-store tx-id))
+  (open-index-snapshot ^java.io.Closeable [this]
+    (db/open-index-snapshot index-store)))
+
+(defn ->node
+  {::sys/args {:db-dir {:doc "Lucene DB Dir"
+                        :required? true
+                        :spec ::sys/path}}}
+  [{:keys [^Path db-dir]}]
+  (let [directory (FSDirectory/open db-dir)
+        analyzer (StandardAnalyzer.)
+        node (LuceneNode. directory analyzer)]
+    (alter-var-root #'*node* (constantly node))))
+
+(defn ->index-store
+  {::sys/deps {:index-store :crux/index-store
```

Comment:
for the `:index-store` default, I don't think you want to refer to the `:crux/index-store` ref - this'll be circular if the user forgets to override it. At first glance, I'd copy the default out of the `:crux/node` component, but may need some further thought - a bigger but cleaner solution may be to explicitly add the concept of 'wrappers' to `crux.system`.